### DISCUSSION TITLE
Mozillaの色んなバージョンのソースコードを読む (let を求めて)

### DISCUSSION THREAD

**Discussion by ubugeeei** [Ubugeeei/work-log#138] [Work Logs]
#137 の続き

**Comment by ubugeeei** [comment]
とりあえずこの辺りをダウンロードして読んでみたけど、この中で一番新しい Mozilla 1.7.13 (2006/4) に let とい言うキーワードはなさそうだった。
![スクリーンショット 2023-08-19 4 04 39](https://github.com/Ubugeeei/work-log/assets/71201308/2ef2757d-91cc-4e39-b71d-e0b611adf59c)

**Comment by ubugeeei** [comment]
https://github.com/Ubugeeei/work-log/discussions/136#discussioncomment-6746833
で教えていただいた通り、let  は JavaScript 1.7で登場している。
https://lia.disi.unibo.it/materiale/JS/developer.mozilla.org/en-US/docs/Web/JavaScript/New_in_JavaScript/1-7.html

じゃあ、これを搭載したMozillaはどのバージョンなのか ?
そこの境界を作る2バージョンのソースコードを読みたい。

**Reply by ubugeeei** [reply]
同リンクに

> The following is a changelog for JavaScript 1.7. This version was included in [Firefox 2](https://developer.mozilla.org/en-US/Firefox/Releases/2) (October 2006).

とあるので 2.0みたい。


これは、https://github.com/Ubugeeei/work-log/discussions/136#discussion-5527764にも記載した Browser compatibility の通り。

**Reply by ubugeeei** [reply]
ついては、2の最初のバージョンと、1の最後のバージョンのソースコードが欲しい。

**Reply by ubugeeei** [reply]
1.7.13 が 2006/4 で、2 が同年の10月だし、アーカイブの中で一番新しいのが 1.7.13 だったので多分 1.7.13 が最後だと思う。

一応調べる。

**Reply by ubugeeei** [reply]
公式にreleasesがあった

https://www.mozilla.org/en-US/firefox/releases/

けど1.5の次が2になっていて、  1.7は？..... と言う気持ちになっている

**Reply by nanto** [reply]
Mozilla 0.6、Mozilla 1.7.13 の「0.6」「1.7.13」というのは Mozilla Application Suite および Gecko レンダリングエンジンのバージョンであり、Firefox のバージョンとは異なります。Firefox 1.0/1.5/2.0 はそれぞれ Gecko 1.7/1.8.0/1.8.1 を搭載しています。参考: [「Gecko - Wikipedia」の「Geckoを利用するソフトウェア 」の項](https://ja.wikipedia.org/wiki/Gecko#Gecko%E3%82%92%E5%88%A9%E7%94%A8%E3%81%99%E3%82%8B%E3%82%BD%E3%83%95%E3%83%88%E3%82%A6%E3%82%A7%E3%82%A2)

**Reply by ubugeeei** [reply]
！！！！
なるほどー！ありがとうございます！スッキリしました！

**Comment by ubugeeei** [comment]
2.0はここからダウンロードできた

https://ftp.mozilla.org/pub/firefox/releases/2.0/source/

**Comment by ubugeeei** [comment]
JSOP_LEAVEBLOCKEXPR というオペコードが登場しているのがわかる。

**Reply by ubugeeei** [reply]
定義は js/src/jsopcode.tbl

**Comment by ubugeeei** [comment]
https://github.com/Ubugeeei/work-log/discussions/138#discussioncomment-6765772

>  1.7.13 だったので多分 1.7.13 が最後だと思う


リリース時期から見るとに、stable はそうなんだが、
old release に 1.8 の alphaとかはあるので、こっちから見てみよう

https://www-archive.mozilla.org/releases/history

**Comment by ubugeeei** [comment]
Firefox 2.0 と 1.x の境目は Gekko 1.8.1 と 1.8.0

https://ja.wikipedia.org/wiki/Gecko#Gecko%E3%82%92%E5%88%A9%E7%94%A8%E3%81%99%E3%82%8B%E3%82%BD%E3%83%95%E3%83%88%E3%82%A6%E3%82%A7%E3%82%A2

**Reply by ubugeeei** [reply]
ちょっと不安なのは、1.8あたり、Mozillaの列に記載がないが、これは別に気にしなくていいんだよな ( ? )

![スクリーンショット 2023-08-19 23 06 39](https://github.com/Ubugeeei/work-log/assets/71201308/444d8e44-5863-45d2-ba20-db0d0b1a4078)

**Reply by ubugeeei** [reply]
一応こちらに Mozilla 1.8 のalpha, beta はある

https://www-archive.mozilla.org/releases/history

1.8 の stable と 1.9が見当たらない

**Reply by ubugeeei** [reply]
old miletone release はこちらから。

https://www-archive.mozilla.org/projects/seamonkey/release-notes/

source code は 13 からしかない ( ? )

**Reply by ubugeeei** [reply]
milestone 13 から const は存在する。
多分ずっと最初からいるんだろうな。
